### DISCUSSION TITLE
Spread Operator in Python

### DISCUSSION THREAD

**Discussion by giscus** [thomasnotfound/how.wtf#56] [Blog Posts]
# Spread Operator in Python

JavaScript provides a spread operator for unpacking elements of iterable objects. The same functionality can be achieved in Python.

https://how.wtf/spread-operator-in-python.html

**Comment by bug-author** [comment]
Comparison with JavaScript was awesome.

**Reply by thomasgtaylor** [reply]
Thank you! I found it best to complement the Python technique with the illustrative JavaScript example.

**Comment by raffasouzaa** [comment]
Thank you, Thomas.. That's exactly what I was looking for! <3

**Reply by thomasgtaylor** [reply]
Awesome! Great to hear!

**Comment by charmzshab** [comment]
Woow!! Exactly what I was looking for. Thanks very much Thomas

**Reply by thomasgtaylor** [reply]
Anytime! Glad you were able to find value!

**Comment by luanmarques1789** [comment]
Grateful explanation, Thomas! Futhermore, what library did you use for code boxes? I liked very much them!

**Reply by thomasgtaylor** [reply]
Sorry for not answering you faster! The highlighting is from [chroma](https://github.com/alecthomas/chroma). It's shown [here](https://github.com/WingLim/hugo-tania/blob/6c4173f56e24a9192398f94d6e71d3e3da6d52de/assets/sass/components/_highlight.scss#L29) in the theme. 👍🏻

**Comment by raul320pl** [comment]
test_obj = { 'foo': 'bar' } is a dict in fact

**Reply by thomasgtaylor** [reply]
You're correct! I was showing a contrasting example in Python. I denoted that above the snippet:
> A very similar technique can be applied with Python dictionaries. Notice the double asterisk operator (**).

**Comment by honzikv** [comment]
Great post. Didn't know you can use spread operator for arguments in JS thats pretty cool.

**Reply by thomasgtaylor** [reply]
Indeed! Thanks for your comment!
